File: Sorting/insertionSort/insertionSort.cpp
/* Insertion Sort implementation in C++
 * Author : Manvi Gupta
 * Input : array length and elements
 * Output : Sorted array elements
 */  

#include <iostream>
using namespace std;

int n;
void insertionSort(int A[])
{
	int key;
  for(int i=0;i<n;i++)
  {
    key = A[i];
    int j = i-1;
    while(j>=0 and key<A[j])
    {
      A[j+1] = A[j];
      j--;
    }
    A[j+1]=key;
    }
}

int main()
{
  std::cout << "Enter the array length: ";
  std::cin >> n;
  int A[n];
  std::cout << "Enter the array elements :" << '\n';
  for(int i=0; i<n; i++)
  cin >> A[i];
  insertionSort(A);
  std::cout << "Sorted Array :" << '\n';
  for(int i=0;i<n;i++)
  cout << A[i] << endl;
  return 0;
}

File: Sorting/Insertion Sort/Python/insertion_sort.py
def insertionSort(arr): 
    for i in range(1, len(arr)): 
        key = arr[i] 
        j = i-1
        while j >=0 and key < arr[j] : 
                arr[j+1] = arr[j] 
                j -= 1
        arr[j+1] = key 

n = int(input())
arr = []
for _ in range(n):
    x = int(input())
    arr.append(x)    

insertionSort(arr) 
for i in range(len(arr)): 
    print ("%d" %arr[i]) 